src/module/actor/creature/document.ts

```diff
@@ -512,6 +512,7 @@ abstract class CreaturePF2e extends ActorPF2e {
                 const updateTracker = args.updateTracker ?? true;
                 if (updateTracker) {
                     game.combat?.setInitiative(combatant.id, roll.total);
+                    await combatant.setFlag("pf2e", "initiativeStatistic", proficiency);
```

Comment:
never use `setFlag`

Reply:
done. should be safe to skip rendering for that update so I added that too.

Reply:
Unfortunately this will only prevent the item sheet from rendering